core/src/main/scala/org/http4s/Method.scala

```diff
@@ -8,44 +8,45 @@ package org.http4s
 
 import cats.implicits._
 import cats.{Eq, Hash, Show}
-import org.http4s.Method.Semantics
 import org.http4s.parser.Rfc2616BasicRules
 import org.http4s.util.{Renderable, Writer}
+import scala.util.hashing.MurmurHash3
 
 /**
   * An HTTP method.
   *
+  * @param name The name of the method
+  *
+  * @param isSafe Request methods are considered "safe" if their defined
+  * semantics are essentially read-only; i.e., the client does not request, and
+  * does not expect, any state change on the origin server as a result of
+  * applying a safe method to a target resource.
+  *
+  * @param isIdempotent A request method is considered "idempotent" if the
+  * intended effect on the server of multiple identical requests with that
+  * method is the same as the effect for a single such request.
+  *
   * @see [http://tools.ietf.org/html/rfc7231#section-4 RFC7321, Section 4]
   * @see [http://www.iana.org/assignments/http-methods/http-methods.xhtml IANA HTTP Method Registry]
   */
-sealed abstract case class Method private (name: String) extends Renderable with Semantics {
+final class Method private (val name: String, val isSafe: Boolean, val isIdempotent: Boolean)
```

Comment:
Side question: technically all Safe HTTP Request's Methods are idempotent, no, @rossabaker?

Note that I'm not proposing any change, but asking this question out of curiosity (after digging through my Github notifications).

Reply:
That's correct.  No public constructor can be safe without also being idempotent.

An alternative and perhaps better design would be to consolidate these two booleans into the old `Semantics` trait, but no longer make it a mixin to the `Method`.